main.tf

```
@@ -13,8 +13,23 @@ resource "aws_config_configuration_recorder" "recorder" {
   count    = module.this.enabled ? 1 : 0
   name     = module.aws_config_label.id
   role_arn = local.create_iam_role ? module.iam_role[0].arn : var.iam_role_arn
+
   recording_group {
-    all_supported                 = true
+    all_supported                 = var.enable_exclusion == true || var.enable_inclusion == true ? false : true
+
+    recording_strategy {
+      use_only = var.enable_exclusion == true ? "EXCLUSION_BY_RESOURCE_TYPES" : (var.enable_inclusion == true ? "INCLUSION_BY_RESOURCE_TYPES" : "ALL_SUPPORTED_RESOURCE_TYPES")
+    }
+
+    # Only if inclusion enabled
+    resource_types = var.enable_inclusion == true && length(var.strategy_resource_types) > 0 ? var.strategy_resource_types : []
```

Looks like this needs to be in `recording_mode` block? based on [this example](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/config_configuration_recorder#periodic-recording) 

also can be simplified
```suggestion
    resource_types = var.enable_inclusion && length(var.strategy_resource_types) > 0 ? var.strategy_resource_types : []
```